connect-widget/app/src/shared/api.ts

```diff
@@ -31,17 +31,30 @@ export function getApiToken(searchParams: URLSearchParams): string {
   return apiToken;
 }
 
-export function handleToken(token: string | null): void {
-  if (!token) {
-    isDemo = true;
-  } else if (isDemoToken(token)) {
+export function handleToken(token: string): void {
+  if (isDemoToken(token)) {
     // tell the user widget in demo mode && disable connect buttons
-    isDemo = true;
     throw new DemoTokenError();
-  } else {
-    // invalid vs valid token logic will go here
-    isDemo = false;
   }
+  if (!isTokenValid()) {
+    // tell the user token is invalid and && disable connect buttons
+    throw new InvalidTokenError();
+  }
+
+  // token is present and valid. Set isDemo to false
+  isDemo = false;
+}
+
+function isTokenValid() {
+  try {
+    getApi().get("/connect/redirect", {
+      params: { provider: "fitbit" },
+    });
```

Comment:
nit: sorry, just noticed it now, but it would be nice to add a quick explanation as to why we're making a call to `/redirect` with provider `fitbit` (I know why, but in a few month's time, I'm 💯 I'll be like "WTF?"... LOL